quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -377,6 +584,37 @@ impl MappingNode {
         }
     }
 
+    /// Finds the field mapping type for a given field path in the mapping tree.
+    /// Dots in `field_path_as_str` define the boundaries between field names.
+    /// If a dot is part of a field name, it must be escaped with '\'.
+    pub fn find_field_mapping_leaf(
+        &mut self,
+        field_path_as_str: &str,
+    ) -> Option<impl Iterator<Item = &mut MappingLeaf>> {
+        let field_path = build_field_path_from_str(field_path_as_str);
+        self.internal_find_field_mapping_leaf(&field_path)
+    }
+
+    fn internal_find_field_mapping_leaf(
+        &mut self,
+        field_path: &[String],
+    ) -> Option<impl Iterator<Item = &mut MappingLeaf>> {
+        let (first_path_fragment, sub_field_path) = field_path.split_first()?;
+        let field_name = self
+            .branches_order
+            .iter()
+            .find(|name| name == &first_path_fragment)?;
+        let child_tree = self.branches.get_mut(field_name).expect("Missing field");
```

Comment:
is it the same as
```suggestion
        let child_tree = self.branches.get_mut(first_path_fragment)?;
```

or can `branches_order` contain a subset of the child nodes?